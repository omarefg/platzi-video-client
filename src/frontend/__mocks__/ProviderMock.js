import React from 'react'
import { createStore, compose, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { Router } from 'react-router'
import { Provider } from 'react-redux'
import { createBrowserHistory } from 'history'
import reducer from '../reducers'
import { initialState } from '../../../db/videos.json'

const store = createStore(reducer, initialState, compose(applyMiddleware(thunk)))
const history = createBrowserHistory()

const ProviderMock = props => {
    const { children } = props

    return (
        <Provider store={store}>
            <Router history={history}>
                {children}
            </Router>
        </Provider>
    )
}

export default ProviderMock
