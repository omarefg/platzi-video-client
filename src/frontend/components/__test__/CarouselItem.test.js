import React from 'react'
import { create } from 'react-test-renderer'
import ProviderMock from '../../__mocks__/ProviderMock'
import { CarouselItem } from '..'

describe('Carousel Item component', () => {
    test('Match Snapshot', () => {
        const carouselItem = create(
            <ProviderMock>
                <CarouselItem/>
            </ProviderMock>,
        )
        expect(carouselItem.toJSON()).toMatchSnapshot()
    })
})
