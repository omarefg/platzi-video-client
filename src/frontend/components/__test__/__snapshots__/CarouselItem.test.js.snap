// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Item component Match Snapshot 1`] = `
<div
  className="carousel-item"
>
  <img
    className="carousel-item__img"
  />
  <div
    className="carousel-item__details"
  >
    <div />
    <p
      className="carousel-item__details--title"
    />
    <p
      className="carousel-item__details--subtitle"
    >
      undefined undefined undefined minutos
    </p>
  </div>
</div>
`;
